pyquil/_parser/PyQuilListener.py

```diff
@@ -242,6 +240,12 @@ def _addr(classical):
     return Addr(int(classical.classicalBit().getText()))
 
 
+def _segment(segment):
+    # type: (QuilParser.SegmentContext) -> Segment
+    ints = segment.INT()
+    return Segment(int(ints[0].getText()), int(ints[1].getText()))
```

Comment:
What if there was only 1 int?

Reply:
I'm not sure (and would have to check against the antlr grammar), but I don't think 1 int counts as a "segment". `[n]` and `[n-n]` play different roles in Quil.

Reply:
Thanks for the catch in the grammar spec: `[n]` was not supposed to be legal segment syntax. I just updated the PR to eliminate this optional case.